src/components/MessagesList/MessagesGroup/Message/MessagePart/Reactions.vue

```diff
@@ -0,0 +1,254 @@
+			:delay="200"
+			:focus-trap="false"
+			:triggers="['hover']"
+			@after-show="fetchReactions">
+			<template #trigger>
+				<NcButton :type="userHasReacted(reaction) ? 'primary' : 'secondary'"
+					class="reaction-button"
+					@click="handleReactionClick(reaction)">
+					{{ reaction }} {{ reactionsCount(reaction) }}
+				</NcButton>
+			</template>
+
+			<div v-if="hasReactions" class="reaction-details">
+				<span>{{ getReactionSummary(reaction) }}</span>
+			</div>
+			<div v-else class="details-loading">
+				<NcLoadingIcon />
+			</div>
+		</NcPopover>
+
+		<!-- More reactions picker -->
+		<NcEmojiPicker v-if="canReact && hasReactions"
+			:per-line="5"
+			:container="`#message_${id}`"
+			@select="handleReactionClick"
+			@after-show="emitEmojiPickerStatus"
+			@after-hide="emitEmojiPickerStatus">
+			<NcButton class="reaction-button"
+				:aria-label="t('spreed', 'Add more reactions')">
+				<template #icon>
+					<EmoticonOutline :size="15" />
+				</template>
+			</NcButton>
+		</NcEmojiPicker>
+	</div>
+</template>
+
+<script>
+import EmoticonOutline from 'vue-material-design-icons/EmoticonOutline.vue'
+
+import { showError } from '@nextcloud/dialogs'
+
+import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
+import NcEmojiPicker from '@nextcloud/vue/dist/Components/NcEmojiPicker.js'
+import NcLoadingIcon from '@nextcloud/vue/dist/Components/NcLoadingIcon.js'
+import NcPopover from '@nextcloud/vue/dist/Components/NcPopover.js'
+
+import { ATTENDEE } from '../../../../../constants.js'
+import { useGuestNameStore } from '../../../../../stores/guestName.js'
+import { useReactionsStore } from '../../../../../stores/reactions.js'
+
+export default {
+	name: 'Reactions',
+
+	components: {
+		NcButton,
+		NcEmojiPicker,
+		NcLoadingIcon,
+		NcPopover,
+		EmoticonOutline,
+	},
+
+	props: {
+		token: {
+			type: String,
+			required: true,
+		},
+		/**
+		 * Whether the current user can react to the message.
+		 */
+		canReact: {
+			type: Boolean,
+			default: false,
+		},
+		/**
+		 * The message id.
+		 */
+		id: {
+			type: [String, Number],
+			required: true,
+		},
+	},
+
+	emits: ['emoji-picker-toggled'],
+
+	setup() {
+		const guestNameStore = useGuestNameStore()
+		const reactionsStore = useReactionsStore()
+		return {
+			guestNameStore,
+			reactionsStore,
+		 }
+	},
+
+	computed: {
+		hasReactions() {
+			return Object.keys(Object(this.detailedReactions)).length !== 0
+		},
+
+		detailedReactions() {
+			return this.reactionsStore.getReactions(this.token, this.id)
+		},
+
+		plainReactions() {
+			return this.$store.getters.message(this.token, this.id).reactions
+		},
+
+		reactionsSelf() {
+			return this.$store.getters.message(this.token, this.id).reactionsSelf
+		},
+
+		/**
+		 * Whether the plain reactions are different than the detailed ones.
+		 */
+		hasMoreReactions() {
+			return this.hasReactions
+					&& Object.keys(this.plainReactions).length !== Object.keys(this.detailedReactions).length
```

Comment:
We need to apply JSON.stringify here somehow, because you could have `{ '🎄': 2, '🔥': 2, '🔒': 2 }` in plain reactions, and `{ '🎄': 3, '🔥': 3, '🔒': 3 }` in detailed => general length is the same, so it passes your check

And I'd call it `hasOutdatedDetails` or something similar

Reply:
`JSON.stringify(this.plainReactions) !== JSON.stringify(this.detailedReactions)` will always be falsy!
plainReactions values are numbers, detailedReactions values - arrays

Reply:
Almost ;)